src/mlang/optimizing_ir/bir_to_oir.ml

```diff
@@ -28,18 +28,14 @@ let initialize_block (bid : Oir.block_id) (blocks : Oir.block Oir.BlockMap.t) :
     Oir.block Oir.BlockMap.t =
   Oir.BlockMap.add bid [] blocks
 
-let rec translate_statement_list (l : Bir.stmt list) (curr_block_id : Oir.block_id)
-    (blocks : Oir.block Oir.BlockMap.t) : Oir.block_id * Oir.block Oir.BlockMap.t =
-  let blocks, last_block_id =
-    List.fold_left
-      (fun (blocks, current_block_id) stmt ->
-        let new_current_block_id, new_blocks = translate_statement stmt current_block_id blocks in
-        (new_blocks, new_current_block_id))
-      (blocks, curr_block_id) l
-  in
-  (last_block_id, blocks)
+let rec translate_statement_list (p : Bir.program) (l : Bir.stmt list)
```

Comment:
`Oir` should contain the concept of rule. Basic blocks in `Oir` should consist of a list of rules that are chained in the control flow.